index.js

```diff
@@ -1,7 +1,7 @@
 'use strict';
 
 var through = require('through2');
-var removeBom = require('remove-bom-buffer');
+var isUTF8 = require('is-utf8');
```

Comment:
Shouldn't we be using https://nodejs.org/dist/latest-v16.x/docs/api/buffer.html#static-method-bufferisencodingencoding ?

Reply:
Since `Buffer.isEncoding` checks an encoding name, it cannot be used instead of `is-utf8`. And it seems that there is no native Node.js function to check utf8 bytes.

Reply:
@sttk you are correct.

Do you think we can use https://nodejs.org/dist/latest-v16.x/docs/api/util.html#new-textdecoderencoding-options with the `ignoreBOM` option instead of this implementation?

Reply:
@phated
In vinyl-fs,  `removeBomStream` is used only when `encoding` is specified. Therefore if this `removeBomStream` function can take `encoding` argument, `is-utf8` could be removed.
https://github.com/gulpjs/vinyl-fs/blob/master/lib/src/read-contents/read-stream.js#L32

About `TextDecoder.decode` with `ignoreBOM`, we could replace the part by `remove-bom-buffer` to it. This would be useful to drop `remove-bom-buffer` in `vinyl-fs`.
https://github.com/gulpjs/vinyl-fs/blob/master/lib/src/read-contents/read-buffer.js#L33

Is it OK to do them?

Reply:
@sttk I like your change! I will review later this week. I also believe that you are correct that vinyl-fs can use TextDecoder in https://github.com/gulpjs/vinyl-fs/blob/master/lib/codecs.js to remove the need for this package.